quickjs/src/nativeMain/kotlin/com/dokar/quickjs/QuickJs.native.kt

```
@@ -347,7 +367,28 @@ actual class QuickJs private constructor(
+            qjs_interrupt_reset(interruptState, evaluationTimeoutMillis)
+            // Cancellation alone can't stop busy JavaScript, so hook it up
+            // to a native interrupt.
+            val interruptOnCancel = Job(coroutineContext.job)
+            interruptOnCancel.invokeOnCompletion { cause ->
+                if (cause is CancellationException && !isClosed) {
+                    qjs_interrupt_request(interruptState)
+                }
+            }
+            try {
+                evalInSession(EvaluationSession(), isRoot = true, block)
+            } catch (e: QuickJsException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)